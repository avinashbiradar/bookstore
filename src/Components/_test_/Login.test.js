import React from 'react';
import {shallow ,mount} from 'enzyme';
import Login from "../LogIn/LogIn"

describe('<Login /> with no props', () => {
  const container = shallow(<Login/>);
  it('should match the snapshot', () => {
    expect(container.html()).toMatchSnapshot();
  });
});

describe('Login Component', () => {
  //testing rendering of login component
    it('should render without throwing an error', () => {
        expect(shallow(<Login/>).exists()).toBe(true)
    })
   // testing the email and password input existence by id
    it('renders a email input', () => {
        expect(shallow( <Login/> ).find('#outlined-email-input').length).toEqual(1)
      })
     it('renders a password input', () => {
        expect(shallow( <Login/> ).find('#outlined-pass-input').length).toEqual(1)
      })
      
   // testing the email and password input existence by id
    it('renders a name input', () => {
        expect(shallow( <Login/> ).find('#outlined-secondary-name').length).toEqual(1)
      })
     it('renders a email input', () => {
        expect(shallow( <Login/> ).find('#outlined-secondary-email').length).toEqual(1)
      })
      it('renders a password input', () => {
        expect(shallow( <Login/> ).find('#outlined-secondary-password').length).toEqual(1)
      })
      // it('renders a password input', () => {
      //   expect(shallow( <Login/> ).find('#outlined-secondary-password').length).toEqual(1)
      // })


    
  
})
// describe('counter testing placeholder ',()=>{
// test("it renders placeholder text same as passed from props",()=>{
//       const placeholderProps="test"
//      const wrapper=shallow(<Login/>)
// //    console.log("wrapper details",wrapper.debug());
//    expect(wrapper.find('Textfield').at(0).props().placeholder).toEqual("Email-input ")
//   })
// })

describe('counter testing ',()=>{
  //testing for specific text in element 
  test('render the title ', () => {
    const wrapper = shallow(<Login/>);
    expect(wrapper.find('span').text()).toContain("ONLINE BOOK SHOPPING");
  })

})

describe("Button Component Tests", () => {
    it("Renders correctly in DOM", () => {
        shallow(
            <button text="Test" />
        );
    });
    it("Expects to find button HTML element in the DOM", () => {
        const wrapper = shallow(<button text="test"/>)
        expect(wrapper.find('button')).toHaveLength(1);
    });

    it("Expects to find button HTML element with className lgn in the DOM", () => {
        const wrapper = shallow(<button className="lgn" text="test"/>)
        expect(wrapper.find('button.lgn')).toHaveLength(1);
    });

    it("Expects to run onClick function when button is pressed in the DOM", () => {
        const mockCallBackClick = jest.fn();
        const wrapper = shallow(<button onClick={mockCallBackClick} className="btn" text="test"/>);
        wrapper.find('button').simulate('click');
        expect(mockCallBackClick.mock.calls.length).toEqual(1);
    });

    it('without component', () => {  
      const ModalWrapperComponent = shallow(<Login />);
      expect(ModalWrapperComponent).toMatchSnapshot();
    });

    it('with component', () => {  
      const props = {
              component: () => {}
           },
           ModalWrapperComponent = shallow(<Login {...props} />);
       expect(ModalWrapperComponent).toMatchSnapshot();
   });

})


// const user = {
//   email: "david@gmail.com",
 
// };
// describe("", () => {
//   it("accepts user account props", () => {
//     const wrapper = mount(<Login user={user} />);
//     expect(wrapper.props().user).toEqual(user);
//   });
//   it("contains users account email", () => {
//     const wrapper = mount(<Login user={user} />);
//     const value = wrapper.find("d").text();
//     expect(value).toEqual("david@gmail.com");
//   });
// });
