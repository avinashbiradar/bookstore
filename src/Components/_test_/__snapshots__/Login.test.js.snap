// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> with no props should match the snapshot 1`] = `"<div class=\\"container-login\\"><div class=\\"img-holder\\"><img class=\\"image\\" alt=\\"image\\"/><span class=\\"text\\">ONLINE BOOK SHOPPING</span></div><div class=\\"login\\"><button class=\\"lgn\\" text=\\"Test\\">LOGIN</button><button class=\\"sgn\\" text=\\"Test\\">SIGN UP</button><div class=\\"active-content\\"><br/><div><div><div class=\\"email\\"><div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined\\" data-shrink=\\"false\\" for=\\"outlined-email-input\\" id=\\"outlined-email-input-label\\">Email</label><div class=\\"MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl\\"><input type=\\"text\\" aria-invalid=\\"\\" id=\\"outlined-email-input\\" name=\\"email\\" value=\\"\\" class=\\"MuiInputBase-input MuiOutlinedInput-input\\"/><fieldset aria-hidden=\\"true\\" class=\\"PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline\\"><legend class=\\"PrivateNotchedOutline-legendLabelled-3\\"><span>Email</span></legend></fieldset></div></div></div><br/><div class=\\"password\\"><div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth\\" endAdornment=\\"[object Object]\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined\\" data-shrink=\\"false\\" for=\\"outlined-pass-input\\" id=\\"outlined-pass-input-label\\">Password</label><div class=\\"MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl\\"><input type=\\"password\\" aria-invalid=\\"\\" id=\\"outlined-pass-input\\" name=\\"password\\" value=\\"\\" class=\\"MuiInputBase-input MuiOutlinedInput-input\\"/><fieldset aria-hidden=\\"true\\" class=\\"PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline\\"><legend class=\\"PrivateNotchedOutline-legendLabelled-3\\"><span>Password</span></legend></fieldset></div></div></div><button text=\\"test\\" type=\\"button\\" class=\\"btn btn btn-primary\\">Login</button></div></div><br/><div class=\\"buttons\\"><button type=\\"button\\" class=\\"btn btn btn-primary\\">Login as Admin</button></div></div><div class=\\"content\\"><div class=\\"MuiFormControl-root MuiTextField-root\\"><label class=\\"MuiFormLabel-root MuiFormLabel-colorSecondary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined\\" data-shrink=\\"false\\" for=\\"outlined-secondary-name\\" id=\\"outlined-secondary-name-label\\">Full Name</label><div class=\\"MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiOutlinedInput-colorSecondary MuiInputBase-formControl\\"><input type=\\"text\\" aria-invalid=\\"\\" id=\\"outlined-secondary-name\\" class=\\"MuiInputBase-input MuiOutlinedInput-input\\"/><fieldset aria-hidden=\\"true\\" class=\\"PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline\\"><legend class=\\"PrivateNotchedOutline-legendLabelled-3\\"><span>Full Name</span></legend></fieldset></div></div><br/><div class=\\"MuiFormControl-root MuiTextField-root\\"><label class=\\"MuiFormLabel-root MuiFormLabel-colorSecondary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined\\" data-shrink=\\"false\\" for=\\"outlined-secondary-email\\" id=\\"outlined-secondary-email-label\\">Email</label><div class=\\"MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiOutlinedInput-colorSecondary MuiInputBase-formControl\\"><input type=\\"text\\" aria-invalid=\\"\\" id=\\"outlined-secondary-email\\" value=\\"\\" class=\\"MuiInputBase-input MuiOutlinedInput-input\\"/><fieldset aria-hidden=\\"true\\" class=\\"PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline\\"><legend class=\\"PrivateNotchedOutline-legendLabelled-3\\"><span>Email</span></legend></fieldset></div></div><br/><div class=\\"MuiFormControl-root MuiTextField-root\\"><label class=\\"MuiFormLabel-root MuiFormLabel-colorSecondary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined\\" data-shrink=\\"false\\" for=\\"outlined-secondary-password\\" id=\\"outlined-secondary-password-label\\">Password</label><div class=\\"MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiOutlinedInput-colorSecondary MuiInputBase-formControl\\"><input type=\\"password\\" aria-invalid=\\"\\" id=\\"outlined-secondary-password\\" value=\\"\\" class=\\"MuiInputBase-input MuiOutlinedInput-input\\"/><fieldset aria-hidden=\\"true\\" class=\\"PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline\\"><legend class=\\"PrivateNotchedOutline-legendLabelled-3\\"><span>Password</span></legend></fieldset></div></div><br/><div class=\\"MuiFormControl-root MuiTextField-root\\"><label class=\\"MuiFormLabel-root MuiFormLabel-colorSecondary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined\\" data-shrink=\\"false\\" for=\\"outlined-secondary-mobile\\" id=\\"outlined-secondary-mobile-label\\">Mobile</label><div class=\\"MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiOutlinedInput-colorSecondary MuiInputBase-formControl\\"><input type=\\"number\\" aria-invalid=\\"\\" id=\\"outlined-secondary-mobile\\" class=\\"MuiInputBase-input MuiOutlinedInput-input\\"/><fieldset aria-hidden=\\"true\\" class=\\"PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline\\"><legend class=\\"PrivateNotchedOutline-legendLabelled-3\\"><span>Mobile</span></legend></fieldset></div></div><button type=\\"button\\" class=\\"btn btn btn-primary\\">Signup</button></div></div></div>"`;

exports[`Button Component Tests with component 1`] = `ShallowWrapper {}`;

exports[`Button Component Tests without component 1`] = `ShallowWrapper {}`;
